cdap-app-fabric/src/main/java/io/cdap/cdap/internal/credential/CredentialIdentityStore.java

```diff
@@ -93,15 +125,15 @@ public Collection<CredentialIdentityId> listForProfile(StructuredTableContext co
    * @throws IOException If any failure reading from storage occurs.
    */
   public Optional<CredentialIdentity> get(StructuredTableContext context, CredentialIdentityId id)
-      throws IOException {
-    StructuredTable table = context.getTable(CredentialProviderStore.CREDENTIAL_IDENTITIES);
-    Collection<Field<?>> key = Arrays.asList(
-        Fields.stringField(CredentialProviderStore.NAMESPACE_FIELD,
-            id.getNamespace()),
-        Fields.stringField(CredentialProviderStore.IDENTITY_NAME_FIELD,
-            id.getName()));
-    return table.read(key).map(row -> GSON.fromJson(row
-        .getString(CredentialProviderStore.IDENTITY_DATA_FIELD), CredentialIdentity.class));
+      throws CipherException, IOException {
+    Optional<StructuredRow> row = readIdentity(context, id);
+    if (!row.isPresent()) {
```

Comment:
Calling isPresent followed by transformation is not considered a good practice. In this case, it's better be:

```java
return readIdentity(context, id)
  .map(r -> r.getBytes(CredentialProviderStore.IDENTITY_DATA_FIELD))
  .map(b -> new String(b, UTF_8))
  .map(s -> GSON.fromJson(s, CredentialIdentity.class));
```

Reply:
Done